cleanlab/rank.py

```diff
@@ -541,3 +542,49 @@ def get_confidence_weighted_entropy_for_each_label(
     label_quality_scores = np.log(label_quality_scores + 1) / clipped_scores
 
     return label_quality_scores
+
+
+def get_knn_distance_ood_scores(
+    query_features: np.array, nbrs: NearestNeighbors, k: int
+) -> np.array:
+    """Returns the KNN distance out-of-distribution (OOD) score for each datapoint.
+
+    This is a function to compute OOD scores where higher scores indicate the datapoint is more likely to be OOD.
+
+    Parameters
+    ----------
+    query_features : np.array
+      Feature matrix of shape ``(N, ...)``, where N is the number of datapoints.
```

Comment:
This would work for higher than 2D data?

Reply:
Good catch. I meant to put shape of (N, M).

I updated docstring with below:
`Feature matrix of shape (N, M), where N is the number of datapoints and M is the number of features.`

Thanks